### ISSUE TITLE
Coordinate text wrapping in navigation toolbar

### ISSUE THREAD

**Issue by jtleighton** [matplotlib/matplotlib#10375]
### Bug report

**Bug summary**

The navigation toolbar at the bottom of a figure window contains a message box for displaying x and y coordinates. With the MacOSX backend, the available space appears to be fixed size and long text wraps and becomes unreadable even though the window width might be able to accommodate more text without wrapping.

**Code for reproduction**

```python
from matplotlib import pyplot as plt
from math import pi, sin

N = 30
x = [n/N for n in range(5*N)]
y = [sin(i*pi) for i in x]

ax = plt.figure().add_subplot(1, 1, 1)
ax.format_coord = lambda x, y : 'the x-coordinate = ' + format(x, '1.4f') + ', ' + \
                                ' and the y-coordinate = ' + format(y, '1.4f')
plt.plot(x, y)
plt.show()
```

**Actual outcome**

<img width="752" alt="small messagebox" src="https://user-images.githubusercontent.com/36115336/35771859-76e1458c-0901-11e8-9def-20ac3430d008.png">

**Expected outcome**

I don't know if the behavior was different in earlier versions. I do know that the behavior is more "sensible" in the TkAgg backend. I haven't tested other backends. In TkAgg the available space expands as the window width expands.

**Matplotlib version**

  * Operating system: Mac OS X 10.13.3 (High Sierra)
  * Matplotlib version: 2.1.1
  * Matplotlib backend (`print(matplotlib.get_backend())`): MacOSX
  * Python version: 3.6.4
  * Jupyter version (if applicable): n/a
  * Other libraries:

Installed Python from installer downloaded from python.org. Installed matplotlib with pip.

**Comment by jtleighton** [comment]
I suspect that this issue is (very understandably) viewed as a minor nuisance issue, and is therefore not likely to be addressed for a long time. I do not have the knowledge to propose an appropriate patch (rectangle width based on window width), but I do have a simple suggestion that would resolve the issue. I suggest changing the fixed rectangle width from 300 to 2500.

`rect.size.width = 2500;`  (See matplotlib/src/_macosx.m: line 1714, branch: master)

In my limited testing, a width of 2500 is adequate for a window that spans the width of a 27" monitor. Better informed people may see problems with this approach that I don't, but I have tested it and am very satisfied with the results. It is an admittedly inelegant solution, but no more inelegant than the current fixed width of 300, and it improves functionality.

I would appreciate any feedback on this suggestion.

**Comment by Vickershhh** [comment]
Hey, how did you change the number of width in _macosx file? I changed it to 1000 and run my code again. But nothing change.... @jtleighton

**Comment by jtleighton** [comment]
Did you rebuild matplotlib? Be sure to follow the instructions for installing from source for your platform. I'll note that, after my initial build, I needed to add the --upgrade option to get pip to rebuild and install matplotlib. See https://matplotlib.org/users/installing.html#installing-from-source

**Comment by timhoffm** [comment]
Is this still an issue on OSX?

**Comment by jklymak** [comment]
Closed by #16710
